homeassistant/components/smartthings/vacuum.py

```diff
@@ -0,0 +1,140 @@
+"""VacuumJet Bot."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pysmartthings import Command
+from pysmartthings.capability import Capability
+
+from homeassistant.components.vacuum import (
+    StateVacuumEntity,
+    VacuumActivity,
+    VacuumEntityFeature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import FullDevice, SmartThingsConfigEntry
+from .const import MAIN
+from .entity import SmartThingsEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+# ───────────────────────────────────────────────
+# Capabilities & Attributes (Const.py?)
+# ───────────────────────────────────────────────
+CAPABILITY_OPERATING_STATE_KEY = "samsungce.robotCleanerOperatingState"  # Const.py?
+ATTRIBUTE_OPERATING_STATE = "operatingState"  # Const.py?
+
+# ───────────────────────────────────────────────
+# Supported Vacuum Features
+# ───────────────────────────────────────────────
+SUPPORTED_FEATURES = (
+    VacuumEntityFeature.TURN_ON
+    | VacuumEntityFeature.TURN_OFF
+    | VacuumEntityFeature.START
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.STATE
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SmartThingsConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Samsung Jet Bot vacuum entities from SmartThings devices."""
+    entry_data = entry.runtime_data
+    vacuums = [
+        SamsungJetBotVacuum(entry_data.client, device)
+        for device in entry_data.devices.values()
+        if isinstance(device.status.get(MAIN), dict)
+        and CAPABILITY_OPERATING_STATE_KEY in device.status[MAIN]
+    ]
+    async_add_entities(vacuums)
+
+
+class SamsungJetBotVacuum(SmartThingsEntity, StateVacuumEntity):
+    """Representation of a Samsung Jet Bot vacuum as a Home Assistant entity."""
+
+    def __init__(self, client, device: FullDevice) -> None:
+        """Initialize the Jet Bot vacuum entity."""
+        super().__init__(client, device, set())
```

Comment:
we need to add the capabilities we want to subscribe to in the set

Reply:
Well, we want to update the state of the device when a capability is updated, and in `set()` we should put all the capabilities we subscribe to which trigger an update